module/VuFind/src/VuFind/Db/Entity/UserEntityInterface.php

```diff
@@ -40,4 +40,47 @@
  */
 interface UserEntityInterface extends EntityInterface
 {
+    /**
+     * Get ID.
+     *
+     * @return int
+     */
+    public function getId(): int;
+
+    /**
+     * Get firstname.
+     *
+     * @return string
+     */
+    public function getFirstname(): string;
+
+    /**
+     * Get lastname.
+     *
+     * @return string
+     */
+    public function getLastname(): string;
+
+    /**
+     * Get last language.
+     *
+     * @return string
+     */
+    public function getLastLanguage(): string;
+
+    /**
+     * Get catalog username.
+     *
+     * @return ?string
+     */
+    public function getCatUsername(): ?string;
+
+    /**
+     * This is a getter for the Catalog Password. It will return a plaintext version
```

Comment:
I think it is better not to include business logic in our entity classes -- my plan is to build a separate service that extracts a usable catalog password from a UserEntityInterface instance, rather than trying to build that directly into the class. Does that make sense to you, or would you prefer to approach this differently? (That was one of the pieces of refactoring I hoped to do once the interface was more fleshed out).